.github/workflows/test.yml

```diff
@@ -0,0 +1,87 @@
+name: test
+
+# Run this workflow on pushes and manually
+on: [push, workflow_dispatch]
+
+jobs:
+  get-go-version:
+    name: "Determine Go toolchain version"
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
+
+  golangci:
+    name: lint
+    needs:
+      - get-go-version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          args: |
+            -v --concurrency 2 \
+            --disable-all \
+            --timeout 10m \
+            --enable gofmt \
+            --enable gosimple \
+            --enable govet
+
+  gomod:
+    needs:
+        - get-go-version
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+      - uses: actions/checkout@v2
+      - name: go mod tidy
+        run: |
+          make mod
+          if ! git diff --exit-code; then
+            echo "Git directory has go.mod or go.sum changes"
+            exit 1
+          fi
+
+  test:
+    needs:
+      - get-go-version
+    runs-on: ubuntu-latest
+    env:
+      TEST_RESULTS_DIR: /tmp/test-results
```

Comment:
I think we can probably remove this part - I'm not sure GHA has an equivalent feature for storing test artifacts like we do in CircleCI.